### PR TITLE
#161044953  User can plan for an approved trip

### PR DESCRIPTION
**Pull request by matthewacha**
#### What does this PR do?
It enables a user to move on with planning for a trip that has been approved

#### Description of Task to be completed?
As a requester, you should be able to begin to plan your trip by providing inputs to the travel checklist items for your destination so that you can become 'Travel Ready'.

#### How should this be manually tested?
* clone this branch `git clone -b ft-plan-approved-trip-161044953  https://github.com/andela/travel_tool_back.git`
*install dependencies `make install`
*start server `make start`
*Navigate to travela in your browser and log in
*Login on your browser and copy the jwt_token from the application tab in Developers tab
*Open Postman
*Add a request and select `POST` method
*Add jwt token in headers with key `Authorization`
*Add `Content-Type` with value `application/json`
*Click on the body option, select raw and choose JSON(application/json) from the drop down
*Add the following data 

##### for uploading file
'{
	"file": <file path>,
	"submissionId":null,
	"tripId": 35678,
	"isUpload": true,
	"data": null,
	"label":"yellowFever"
}'

#### for no upload
'{
	"file": null,
	"submissionId":null,
	"tripId": 35678,
	"isUpload": false,
    "data":{
        "airline":"Kenya Airways",
        "flightNumber":"KQ 532",
        "arrivalTime":"19:35:00"
    },
    "label":"AirTicket"
}'

*Send a `POST` request to the endpoint using `/api/v1/checklists/:checkListItemId/submission`

#### What are the relevant pivotal tracker stories?
[61044953](https://www.pivotaltracker.com/story/show/161044953)

##### file upload
image

##### non upload
image

### CODE REVIEW

File: src/modules/travelChecklist/TravelChecklistController.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    let submission;
+    const extension = file ? file.split('.')[1] : null;
+    const body = { tripId, checklistItemId };
+    if (isUpload) {
+      try {
+        checklistSubmission = await TravelChecklistController
+          .handleCloudinaryUpload(file, [extension, label], body);
+      } catch (error) {
+        if (error.error) {
+          const isError = error.error.errno === -2 ? error : '';
```
Comment (johnmutuma5) [inline src/modules/travelChecklist/TravelChecklistController.js]: @matthewacha, What is `errorno -2`?

Comment (matthewacha) [reply src/modules/travelChecklist/TravelChecklistController.js]: @johnmutuma5 it is an error number in the object returned  by cloudinary when it fails to upload the file

Comment (johnmutuma5) [reply src/modules/travelChecklist/TravelChecklistController.js]: @matthewacha, okay.

**Comment by Spectrumsun** [comment]
@matthewacha  good work. but why did you decide to upload the image in the backend instead of uploading from the front end? which will reduce the amount of resources sent to the backend

**Comment by samuelkubai** [comment]
@matthewacha why have you allowed this story to run stale?

File: .env.example [hunk 2]
```diff
@@ -9,9 +9,18 @@ REDIRECT_URL=frontend baseurl
 MAILGUN_API_KEY=mailgun api key
 MAILGUN_DOMAIN_NAME=mailgun domain name
 MAIL_SENDER=travel sender mail
+<<<<<<< HEAD
```
Comment (ezrqnkemboi) [inline .env.example]: Good work @matthewacha . Just look at this file, it needs to remove these conflicts.

File: src/modules/travelChecklist/TravelChecklistController.js [hunk 3]
```diff
@@ -226,4 +227,109 @@ export default class TravelChecklistController {
       CustomError.handleError(error.stack, 500, res);
     }
   }
+
+  static async handleCloudinaryUpload(file, tags, props, fileName) {
+    const { tripId, checklistItemId } = props;
```
Comment (ezrqnkemboi) [inline src/modules/travelChecklist/TravelChecklistController.js]: I actually like the idea that you used the cloudinary SaaS for managing assets.

File: .env.example [hunk 4]
```diff
@@ -9,9 +9,18 @@ REDIRECT_URL=frontend baseurl
...
 CLOUDINARY_CLOUD_NAME='travela'
 CLOUDINARY_API_KEY=
 CLOUDINARY_API_SECRET=
 CLOUDINARY_ENHANCE_IMAGE=true
 CLOUDINARY_STATIC_IMAGE=false
-APP_EMAIL=
\ No newline at end of file
+APP_EMAIL=
+=======
```
Comment (ezrqnkemboi) [inline .env.example]: Check above comment @matthewacha .

**Pull request merged**